### PR TITLE
[Build settings] add "missing newline at end of file" warning

### LINKED ISSUE #2080: Add newline at the end of all source files that miss that.
-Wnewline-eof  will complain while compiling code that include ASDK headers. It more readable if we can add new lines to each of those files. Especially, to the following files:
/AsyncDisplayKit/ASMapNode.h
/AsyncDisplayKit/ASRunLoopQueue.h
/AsyncDisplayKit/ASViewController.h
/AsyncDisplayKit/Details/ASEnvironment.h
/AsyncDisplayKit/Details/ASMutableAttributedStringBuilder.h
/AsyncDisplayKit/Details/ASPhotosFrameworkImageRequest.h
/AsyncDisplayKit/Details/ASRangeController.h
/AsyncDisplayKit/Details/NSMutableAttributedString+TextKitAdditions.h
/AsyncDisplayKit/Details/UICollectionViewLayout+ASConvenience.h
/AsyncDisplayKit/Layout/ASStackLayoutSpec.h
(the list is based on an old ASDK cut, so it's not meant to be updated/complete)

### PR DESCRIPTION
**Pull request by hannahmbanana**
Fixes #2080.

### CODE REVIEW

**Comment by hannahmbanana** [comment]
It looks like we should enable a few more warnings. I think we'll be looking at this tomorrow morning, right?

File: AsyncDisplayKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -438,7 +438,7 @@
 		DE4843DB1C93EAB100A1F33B /* ASLayoutTransition.mm in Sources */ = {isa = PBXBuildFile; fileRef = E52405B21C8FEF03004DC8E7 /* ASLayoutTransition.mm */; };
 		DE4843DC1C93EAC100A1F33B /* ASLayoutTransition.h in Headers */ = {isa = PBXBuildFile; fileRef = E52405B41C8FEF16004DC8E7 /* ASLayoutTransition.h */; };
 		DE6EA3231C14000600183B10 /* ASDisplayNode+FrameworkPrivate.h in Headers */ = {isa = PBXBuildFile; fileRef = DE6EA3211C14000600183B10 /* ASDisplayNode+FrameworkPrivate.h */; };
-		DE72E2AC1D4D90E000DDB258 /* ASLayoutSpec+Private.h in CopyFiles */ = {isa = PBXBuildFile; fileRef = 69F6058C1D3DA27E00C8CA38 /* ASLayoutSpec+Private.h */; };
+		DE72E2AC1D4D90E000DDB258 /* (null) in CopyFiles */ = {isa = PBXBuildFile; };
```
Comment (maicki) [inline AsyncDisplayKit.xcodeproj/project.pbxproj]: Any idea why we have a `(null)` in here?

File: AsyncDisplayKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -709,7 +709,7 @@
 				F7CE6C611D2CDB3E00BE4C15 /* ASLayoutable.h in CopyFiles */,
 				F7CE6C621D2CDB3E00BE4C15 /* ASLayoutablePrivate.h in CopyFiles */,
 				F7CE6C631D2CDB3E00BE4C15 /* ASLayoutSpec.h in CopyFiles */,
-				DE72E2AC1D4D90E000DDB258 /* ASLayoutSpec+Private.h in CopyFiles */,
+				DE72E2AC1D4D90E000DDB258 /* (null) in CopyFiles */,
```
Comment (maicki) [inline AsyncDisplayKit.xcodeproj/project.pbxproj]: Any idea why we have a `(null)` in here?

File: AsyncDisplayKit.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -2013,6 +2016,21 @@
 			shellScript = "\"${SRCROOT}/Pods/Target Support Files/Pods-AsyncDisplayKitTests/Pods-AsyncDisplayKitTests-resources.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
+		47FEBAC527C86C3C7E855E41 /* 📦 Embed Pods Frameworks */ = {
```
Comment (maicki) [inline AsyncDisplayKit.xcodeproj/project.pbxproj]: You are probably on CocoaPods version 1.0 and you should update to 1.0.1 so we do not add the emojis back in to the project settings. Not sure if we have to add these kind of settings at all ...

**Comment by maicki** [comment]
@hannahmbanana +1 For adding more warnings. Added a couple of comments but overall looks good.

**Comment by keleftheriou** [comment]
Can one of the admins verify this patch?

**Comment by hannahmbanana** [comment]
Cocoapod issues. ugh. Closing this one in favor of #2103.

**Pull request closed without merging**